plugins/base/src/main/kotlin/translators/descriptors/DescriptorAccessorConventionUtil.kt

```diff
@@ -0,0 +1,83 @@
+import org.jetbrains.kotlin.load.java.propertyNameByGetMethodName
+import org.jetbrains.kotlin.load.java.propertyNamesBySetMethodName
+
+internal data class DescriptorFunctionsHolder(
+    val regularFunctions: List<FunctionDescriptor>,
+    val accessors: Map<PropertyDescriptor, List<FunctionDescriptor>>
+)
+
+internal fun splitFunctionsAndAccessors(
+    properties: List<PropertyDescriptor>,
+    functions: List<FunctionDescriptor>
+): DescriptorFunctionsHolder {
+    val propertiesByName = properties.associateBy { it.name.asString() }
+    val regularFunctions = mutableListOf<FunctionDescriptor>()
+    val accessors = mutableMapOf<PropertyDescriptor, MutableList<FunctionDescriptor>>()
+    functions.forEach { function ->
+        val possiblePropertyNamesForFunction = function.toPossiblePropertyNames()
+        val property = possiblePropertyNamesForFunction.firstNotNullOfOrNull { propertiesByName[it] }
+        if (property != null && function.isAccessorFor(property)) {
+            accessors.getOrPut(property, ::mutableListOf).add(function)
+        } else {
+            regularFunctions.add(function)
+        }
+    }
+    return DescriptorFunctionsHolder(regularFunctions, accessors)
+}
+
+internal fun FunctionDescriptor.toPossiblePropertyNames(): List<String> {
+    val stringName = this.name.asString()
+    return when {
+        JvmAbi.isSetterName(stringName) -> propertyNamesBySetMethodName(this.name).map { it.asString() }
+        JvmAbi.isGetterName(stringName) -> propertyNamesByGetMethod(this)
+        else -> listOf()
+    }
+}
+
+internal fun propertyNamesByGetMethod(functionDescriptor: FunctionDescriptor): List<String> {
+    val stringName = functionDescriptor.name.asString()
+    // In java, the convention for boolean property accessors is as follows:
+    // - `private boolean active;`
+    // - `private boolean isActive();`
+    //
+    // Whereas in Kotlin, because there are no explicit accessors, the convention is
+    // - `val isActive: Boolean`
+    //
+    // This makes it difficult to guess the name of the accessor property in case of Java
+    val javaPropName = if (functionDescriptor is JavaMethodDescriptor && JvmAbi.startsWithIsPrefix(stringName)) {
+        val javaPropName = stringName.removePrefix("is").let { newName ->
+            newName.replaceFirst(newName[0], newName[0].toLowerCase())
+        }
+        javaPropName
+    } else {
+        null
+    }
+    val kotlinPropName = propertyNameByGetMethodName(functionDescriptor.name)?.asString()
+    return listOfNotNull(javaPropName, kotlinPropName)
+}
+
+internal fun FunctionDescriptor.isAccessorFor(property: PropertyDescriptor): Boolean {
+    return this.isGetterFor(property) || this.isSetterFor(property)
+}
+
+internal fun FunctionDescriptor.isGetterFor(property: PropertyDescriptor): Boolean {
+    return this.returnType == property.returnType
+            && this.valueParameters.isEmpty()
+            && !property.visibility.isPublicAPI
```

Comment:
Do we have a test of this logic (there is a non-public property)?

Reply:
Yeah, Andrej wrote a test for that and it failed when I forgot to implement it :) 

The name is `kotlin inheriting java should not append anything since field is public` inside `DescriptorSuperPropertiesTest`